ksqldb-tools/src/test/java/io/confluent/ksql/tools/migrations/commands/ApplyMigrationCommandTest.java

```diff
@@ -246,6 +245,34 @@ public void shouldApplyDefineUndefineCommands() throws Exception {
     inOrder.verifyNoMoreInteractions();
   }
 
+  @Test
+  public void shouldResetVariablesBetweenMigrations() throws Exception {
```

Comment:
Thanks for the extra test coverage! This is great. Could we also add an equivalent test to verify that SET/UNSET are also scoped to a single migrations file? (Feel free to defer to a follow-up PR if that's more convenient.)